static/app/components/smartSearchBar/index.tsx

```diff
@@ -1671,7 +1776,17 @@ class SmartSearchBar extends Component<Props, State> {
           )}
         </ActionsBar>
 
-        {this.state.showDropdown && (
+        {this.shouldShowDatePicker && (
+          <SearchBarDatePicker
+            date={this.cursorValueIsoDate?.value}
+            dateString={this.cursorValueIsoDate?.text}
+            handleSelectDateTime={value => {
+              this.onAutoCompleteIsoDate(value);
+            }}
+          />
+        )}
+
+        {this.state.showDropdown && !this.shouldShowDatePicker && (
```

Comment:
Maybe extract this into a getter functon `shouldShowAutocomplete` and keep it next to `shouldShowDatePicker`?